docs: "product" typo in CartProvider docs

jh3y opened the pull request:
Fix teeny typo in `CartProvider` docs. Also, remove `yarn-error.log` unless it's being kept for a reason.
I can't replicate this locally. But, currently the "Edit this page on Github" links are pointing to the wrong place. Might be worth a docs push?

![Linking to wrong place](https://user-images.githubusercontent.com/842246/108512067-9d16d400-72b8-11eb-83d4-e53289b95fe0.png)

dayhaysoos commented:
Thanks @jh3y ! I'm gonna merge this. I plan on redoing the docs site completely soon, didn't know they pointed to the wrong place, though. Gonna take a look at that this weekend

jh3y commented:
No worries 👍 

I think it might fix itself if the docs get redeployed. I couldn't replicate it locally. Might have been a version change since the last deploy. I did see the issue reported on `gatsby-theme-docs` 👍  

Hopefully, will be a non-issue.

Great work with this repo. Looking forward to taking it for a test drive soon! 💯
